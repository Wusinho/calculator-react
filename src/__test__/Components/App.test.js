import { create } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import App from '../../components/app';

test('renders correctly', () => {
  const tree = create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('displays heading', () => {
  render(<App />);
  const p = screen.getByText("Let's do some maths");
  expect(p).toBeInTheDocument();
});