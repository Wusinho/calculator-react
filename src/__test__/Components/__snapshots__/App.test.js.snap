// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ui grid container"
>
  <div
    className="columns is-mobile is-centered"
  >
    <div
      className="column"
    >
      <p
        className="title has-text-white"
      >
        Let's do some maths 
      </p>
    </div>
    <div
      className="column"
    >
      <div
        className="calContainer"
      >
        <h1
          className="display-value"
        />
        <div
          className="d-grid"
        >
          <button
            className="ui button purple"
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
          <button
            className="ui button purple"
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
          <button
            className="ui button purple"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
          <button
            className="ui button blue"
            onClick={[Function]}
            type="button"
          >
            /
          </button>
        </div>
        <div
          className="d-grid"
        >
          <button
            className="ui button teal"
            onClick={[Function]}
            type="button"
          >
            7
          </button>
          <button
            className="ui button teal"
            onClick={[Function]}
            type="button"
          >
            8
          </button>
          <button
            className="ui button teal"
            onClick={[Function]}
            type="button"
          >
            9
          </button>
          <button
            className="ui button blue"
            onClick={[Function]}
            type="button"
          >
            X
          </button>
        </div>
        <div
          className="d-grid"
        >
          <button
            className="ui button teal"
            onClick={[Function]}
            type="button"
          >
            4
          </button>
          <button
            className="ui button teal"
            onClick={[Function]}
            type="button"
          >
            5
          </button>
          <button
            className="ui button teal"
            onClick={[Function]}
            type="button"
          >
            6
          </button>
          <button
            className="ui button blue"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
        </div>
        <div
          className="d-grid"
        >
          <button
            className="ui button teal"
            onClick={[Function]}
            type="button"
          >
            1
          </button>
          <button
            className="ui button teal"
            onClick={[Function]}
            type="button"
          >
            2
          </button>
          <button
            className="ui button teal"
            onClick={[Function]}
            type="button"
          >
            3
          </button>
          <button
            className="ui button blue"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </div>
        <div
          className="d-grid-last"
        >
          <button
            className="ui button teal"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
          <button
            className="ui button teal"
            onClick={[Function]}
            type="button"
          >
            .
          </button>
          <button
            className="ui button blue"
            onClick={[Function]}
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
